src/rpc/server.cpp

```diff
@@ -360,11 +360,24 @@ bool IsDeprecatedRPCEnabled(const std::string& method)
 
 UniValue JSONRPCExec(const JSONRPCRequest& jreq)
 {
-    // Might throw exception. Single requests will throw and send HTTP error codes
-    // but inside a batch, we just include the error object and return HTTP 200
-    UniValue result = tableRPC.execute(jreq);
+    UniValue result;
+    if (jreq.m_json_version == JSONVersion::JSON_2_0) {
+        // JSONRPC 2.0 behavior: only throw HTTP error if the server is actually
+        // broken. Otherwise errors are sent back in "HTTP OK" responses.
+        try {
+            result = tableRPC.execute(jreq);
+        } catch (const UniValue& objError) {
+            return JSONRPCReplyObj(NullUniValue, objError, jreq.id, jreq.m_json_version);
```

Comment:
In commit "rpc: make error and result mutually exclusive in JSON-RPC 2.0 replies" (c9196a2e5b747753e0d5ee59e1bcb987943eedbf)

Again suggestion for an separate earlier commit, but it would be good if JSONRPCReplyObj took a `UniValue error` parameter instead of `const UniValue& error` so the error object could be moved instead of copied.

Reply:
I updated the function `JSONRPCReplyObj` to expect plain ol' `UniValue` for both response and error, and added `std::move()` where it seemed applicable. Tidy didn't like `std::move()` here because `objError` is already a `const`, so my understanding is that the error message will continue to be passed by reference through the end of `JSONRPCReplyObj` but my understanding is limited